homeassistant/components/steam_online/sensor.py

```diff
@@ -28,28 +31,42 @@
         vol.All(cv.ensure_list, [cv.string]),
 })
 
+APP_LIST_KEY = 'steam_online.app_list'
+BASE_INTERVAL = timedelta(minutes=1)
+
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Steam platform."""
     import steam as steamod
     steamod.api.key.set(config.get(CONF_API_KEY))
     # Initialize steammods app list before creating sensors
     # to benefit from internal caching of the list.
-    steam_app_list = steamod.apps.app_list()
-    add_entities(
-        [SteamSensor(account,
-                     steamod,
-                     steam_app_list)
-         for account in config.get(CONF_ACCOUNTS)], True)
+    hass.data[APP_LIST_KEY] = steamod.apps.app_list()
+    entities = [
+        SteamSensor(account, steamod)
+        for account in config.get(CONF_ACCOUNTS)]
+    if not entities:
+        return
+    add_entities(entities, True)
+
+    # Only one sensor update once every 60 seconds to avoid
+    # flooding steam and getting disconnected.
+    entity_next = 0
+    @callback
+    def do_update(time):
+        nonlocal entity_next
+        entities[entity_next].async_schedule_update_ha_state(True)
+        entity_next = (entity_next + 1) % len(entities)
```

Comment:
There seems to be a way to batch fetch profiles in fewer (single) requests.

https://github.com/Lagg/steamodd/blob/2e9ced4e7a6dbe3e09d5a648450bafc12b937b95/steam/user.py#L290

Then we could make a single api class that interacts with the api, caches the data and signals the entities to update when new data is ready.

Reply:
Could do, I don't actually use this component so I don't know how many profiles are actually gonna get set up, or even if batching is actually considered "one request".

Reply:
I think it's a good idea, but let's not make it a requirement for this PR.